gulpfile.js

```diff
@@ -1,237 +1,24 @@
-    'node_modules/angular-highlightjs/angular-highlightjs.js',
-    'node_modules/moment/min/moment.min.js',
-    'node_modules/angular-moment/angular-moment.js',
-    'node_modules/foundation-apps/js/vendor/**/*.js',
-    'node_modules/foundation-apps/js/angular/**/*.js',
-    '!node_modules/foundation-apps/js/angular/app.js'
-  ],
-}
-
-// 3. TASKS
-// - - - - - - - - - - - - - - -
-
-// Cleans the build directory
-gulp.task('clean', function(cb) {
-  rimraf('./dist', cb);
-});
-
-// Copies everything in the client folder except templates, Sass, and JS
-gulp.task('copy', function() {
-  return gulp.src(paths.staticAssets, {
-    base: './src/'
-  })
-    .pipe(gulp.dest('./dist'))
-  ;
-});
-gulp.task('copy-settings', function() {
-  return gulp.src('./src/settings.js')
-    .pipe(gulp.dest('./dist/assets/js/settings/'))
-  ;
-});
-
-// Copies your app's page templates and generates URLs for them
-gulp.task('copy:templates', function() {
-  return gulp.src(paths.templates)
-    .pipe(router({
-      path: 'dist/assets/js/routes.js',
-      root: 'src'
-    }))
-    .pipe(gulp.dest('./dist/templates'))
-  ;
-});
-
-// Compiles Sass
-gulp.task('sass', function () {
-  var minifyCss = $.if(isProduction, $.minifyCss());
-
-  return gulp.src('src/assets/scss/app.scss')
-    .pipe($.sass({
-      includePaths: paths.css,
-      outputStyle: (isProduction ? 'compressed' : 'nested'),
-      errLogToConsole: true
-    }))
-    .pipe($.autoprefixer({
-      browsers: ['last 2 versions', 'ie 10']
-    }))
-    .pipe(minifyCss)
-    .pipe(gulp.dest('./dist/assets/css/'))
-  ;
-});
-
-// Compiles and copies the Foundation for Apps JavaScript, as well as your app's custom JS
-gulp.task('uglify', ['uglify:foundation', 'uglify:app'])
-
-// Compiles the Foundation for Apps directive partials into a single JavaScript file
-gulp.task('copy:foundation', function (cb) {
-  gulp.src('node_modules/foundation-apps/js/angular/components/**/*.html')
-    .pipe($.ngHtml2js({
-      prefix: 'components/',
-      moduleName: 'foundation',
-      declareModule: false
-    }))
-    .pipe($.uglify())
-    .pipe($.concat('foundation_templates.js'))
-    .pipe(gulp.dest('./dist/assets/js'))
-    ;
-
-  // Iconic SVG icons
-  gulp.src('./node_modules/foundation-apps/iconic/**/*')
-    .pipe(gulp.dest('./dist/assets/images/iconic/'))
-    ;
-
+const gulp = require('gulp');
+require('require-dir')('gulp_tasks');
+const browserSync = require('browser-sync');
+
+const conf = require('./conf/gulp.conf');
+
+gulp.task('watch', watch);
+gulp.task('build', gulp.series(gulp.parallel('other', 'webpack:dist')));
+gulp.task('build:dev', gulp.series(gulp.parallel('other', 'webpack:dev')));
+gulp.task('default', gulp.series('build'));
+gulp.task('test', gulp.series('karma:single-run'));
+gulp.task('test:auto', gulp.series('karma:auto-run'));
+gulp.task('serve', gulp.series('webpack:watch', 'watch', 'browsersync'));
+gulp.task('serve:dist', gulp.series('default', 'browsersync:dist'));
```

Comment:
Much more manageable gulp tasks (IMHO). A lot of the gulp plugins are replace by webpack.